lib/mix_unused/analyzers/unreachable.ex

```diff
@@ -0,0 +1,49 @@
+defmodule MixUnused.Analyzers.Unreachable do
+  @moduledoc """
+  Finds all the reachable exported functions starting from a set of well-known used functions.
+  All remaining functions are considered "unused".
+  """
+
+  alias MixUnused.Analyzers.Calls
+  alias MixUnused.Analyzers.Unreachable.Config
+  alias MixUnused.Analyzers.Unreachable.Usages
+  alias MixUnused.Meta
+
+  @behaviour MixUnused.Analyze
+
+  @impl true
+  def message, do: "is unreachable"
+
+  @impl true
+  def analyze(data, exports, config) do
+    config = Config.cast(config)
+    graph = Calls.calls_graph(data, exports)
+    usages = Usages.usages(config, exports)
+    reachables = graph |> Graph.reachable(usages) |> MapSet.new()
+    called_at_compile_time = Calls.called_at_compile_time(data, exports)
+
+    for {mfa, _meta} = call <- exports,
+        filter_transitive_call?(config, graph, mfa),
```

Comment:
Penso di aver capito il giro del fumo su questo tool, ma alcune zone sono rimaste oscure, tipo questa come mai abbiamo bisogno di trovare le transitive call ?

Reply:
questo è un filtering che facciamo in base all'opzione `report_transitively_unused` per fare il report delle sole radici del grafo delle chiamate che potrebbero risultare non utilizzate.  Il tool altrimenti farebbe il report di ogni chiamata non "reachable". Serve a facilitare il processo di individuazione degli entrypoint